crest/Assets/Crest/Crest/Scripts/Shapes/FFT/ShapeFFT.cs

```diff
@@ -0,0 +1,422 @@
+using UnityEditor;
+#endif
+
+namespace Crest
+{
+    /// <summary>
+    /// FFT ocean wave shape
+    /// </summary>
+    [ExecuteAlways]
+    [AddComponentMenu(Internal.Constants.MENU_PREFIX_SCRIPTS + "Shape FFT")]
+    [HelpURL("https://crest.readthedocs.io/en/latest/user/wave-conditions.html#shapefft")]
+    public partial class ShapeFFT : MonoBehaviour, LodDataMgrAnimWaves.IShapeUpdatable
+#if UNITY_EDITOR
+        , IReceiveSplinePointOnDrawGizmosSelectedMessages
+#endif
+    {
+        [Header("Wave Conditions")]
+        [Tooltip("Impacts how aligned waves are with wind.")]
+        [Range(0, 1)]
+        public float _windTurbulence = 0.145f;
+        [Tooltip("Choppiness of waves.")]
+        [Range(0, 2.5f)]
+        public float _chop = 1f;
+
+        [Tooltip("The spectrum that defines the ocean surface shape. Assign asset of type Crest/Ocean Waves Spectrum."), Embedded]
+        public OceanWaveSpectrum _spectrum;
+        OceanWaveSpectrum _activeSpectrum = null;
+
+        [Tooltip("When true, the wave spectrum is evaluated once on startup in editor play mode and standalone builds, rather than every frame. This is less flexible but reduces the performance cost significantly."), SerializeField]
+        bool _spectrumFixedAtRuntime = true;
+
+        [Tooltip("Primary wave direction heading (deg). This is the angle from x axis in degrees that the waves are oriented towards. If a spline is being used to place the waves, this angle is relative ot the spline."), Range(-180, 180)]
+        public float _waveDirectionHeadingAngle = 0f;
+        public Vector2 PrimaryWaveDirection => new Vector2(Mathf.Cos(Mathf.PI * _waveDirectionHeadingAngle / 180f), Mathf.Sin(Mathf.PI * _waveDirectionHeadingAngle / 180f));
+
+        [Tooltip("When true, uses the wind speed on this component rather than the wind speed from the Ocean Renderer component.")]
+        public bool _overrideGlobalWindSpeed = false;
+        [Tooltip("Wind speed in km/h. Controls wave conditions."), Range(0, 150f), Predicated("_overrideGlobalWindSpeed")]
+        public float _windSpeed = 20f;
+
+        [Tooltip("Multiplier for these waves to scale up/down."), Range(0f, 1f)]
+        public float _weight = 1f;
+
+        [Tooltip("How much these waves respect the shallow water attenuation setting in the Animated Waves Settings. Set to 0 to ignore shallow water."), SerializeField, Range(0f, 1f)]
+        float _respectShallowWaterAttenuation = 1f;
+
+        [HideInInspector]
+        [Delayed, Tooltip("How many wave components to generate in each octave.")]
+        public int _componentsPerOctave = 8;
+
+        [HideInInspector]
+        [Tooltip("Change to get a different set of waves.")]
+        public int _randomSeed = 0;
+
+        [Header("Generation Settings")]
+        [Tooltip("Resolution to use for wave generation buffers. Low resolutions are more efficient but can result in noticeable patterns in the shape."), Delayed]
+        public int _resolution = 32;
```

Comment:
If I change resolution, the shape looks incorrect until I disable/enable the component. It might need something like the auto-rebuild that OceanRenderer has?

Reply:
thanks fixed now, was not re-init'ing the res properly